guides/dimensiondata.md

```diff
@@ -0,0 +1,261 @@
+
+## <a id="build-dimension-data-cloud-control-api"></a>How to: Build the Dimension Data Cloud Control API
+
+{% highlight java%}
+ApiContext<DimensionDataCloudControlApi> ctx = null;
+try
+{
+    ctx = ContextBuilder.newBuilder("dimensiondata-cloudcontrol")
+            .endpoint(endpoint)
+            .credentials(username, password)
+            .modules(ImmutableSet.of(new SLF4JLoggingModule()))
+            .build();
+
+    /*
+     * Retrieve the Guice injector from the context.
+     * We will use this for retrieving the some Predicates that are used by the following operations.
+     */
+    Injector injector = ctx.utils().injector();
+    DimensionDataCloudControlApi api = ctx.getApi();
+
+    /*
+     * interact with the DimensionDataCloudControlApi
+     */
+}
+finally
+{
+    if (ctx != null)
+    {
+        ctx.close();
+    }
+}
+{% endhighlight %}
+
+# How To's
+## <a id="how-to-list-a-data-center"></a>How to: List a Data Center
+
+The user account will be associated with one or more [MCP 2.0 Data Centers](https://docs.mcp-services.net/display/CCD/Introduction+to+MCP+2.0+Data+Center+Locations). A Data Center is a container for the assets that will be deployed and created by the jclouds Dimension Data CloudControl Provider.
+
+In jclouds terminology a Zone equates to a Data Center.
+
+The following code example shows you how to programmatically list Data Centers: 
+
+{% highlight java %}
+PagedIterable<Datacenter> datacenters = api.getInfrastructureApi().listDatacenters();
+{% endhighlight %}
+
+This responds with a paged result set containing the Data Centers visible to your organization.
+
+## <a id="how-to-deploy-a-network-domain"></a>How to: Deploy a Network Domain
+
+For more information on Network Domains see [here](https://docs.mcp-services.net/display/CCD/Introduction+to+Cloud+Network+Domains+and+VLANs).
+
+The code to deploy a Network Domain follows.
+
+{% highlight java %}
+/*
+ * Deploy Network Domain to the Zone / Datacenter we wish to operate on. The response from this API is the Network Domain Identifier.
+ */
+String networkDomainId = api.getNetworkApi().deployNetworkDomain(AU9, "jclouds-example", "jclouds-example", "ESSENTIALS");
+{% endhighlight %}
+
+A Network Domain deployment is an asynchronous process. We need to wait for it to complete. The Dimension Data provider
+has built in google guice predicates that will block execution and check that the Network Domain's State has moved from `PENDING_ADD` to `NORMAL`.
+
+The following is some example code that shows how the to use predicate suitable for asserting a Network Domain state has transitioned to the `NORMAL` state. The predicate uses the Network Domain Identifier we wish to check the state of.
+{% highlight java %}
+Predicate<String> networkDomainNormalPredicate = injector.getInstance(Key.get(new TypeLiteral<Predicate<String>>()
+{
+}, Names.named(NETWORK_DOMAIN_NORMAL_PREDICATE)));
```

Comment:
My intention with the comment was to make this more usable by extracting the whole `Key` object into a constant, not just the string. This way users wouldn't need to build this ugly construct.

However, if it makes sense to expose this predicate to users, then it's better to add the following method to the `NetworkApi` interface:
```java
@Provides
@Named("NETWORK_DOMAIN_NORMAL_PREDICATE")
Predicate<String> networkDomainNormalPredicate();
```

This way useres can jsut get it with `api. getNetworkApi().networkDomainNormalPredicate()` (the `@Provides` annotation does the magic here), which makes much more sense and is more usable. Apply the same pattern to the other user-facing predicates and their corresponding API interfaces.

Feel free to open a PR in the provider to add these methods, but let's change the docs now and move forward.

Reply:
@nacx I understand. In fact our plan was to make this change during the past week or two but ran out of time (we have an internal jira for it). I will adjust the docs to show what this will eventually look like. Appreciate your time on this PR.

One question - is there a planned release date for 2.2.0?